### ISSUE TITLE
RRTMG with ECMWF aerosols 

### ISSUE THREAD

**Issue by wenshanw** [climlab/climlab#79]
having trouble running RRTM with aerosols.
I used iaer = 6 and defined ecaer as an arrary of (types_of_aerosols, ncol, nlayer).

Error msg:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-33-246e8494651c> in <module>()
----> 1 rad.compute_diagnostics()

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/process/time_dependent_process.py in compute_diagnostics(self, num_iter)
    344         """
    345         for n in range(num_iter):
--> 346             ignored = self.compute()
    347 
    348     def _update_time(self):

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/process/time_dependent_process.py in compute(self)
    205         tendencies = {}
    206         ignored = self._compute_type('diagnostic')
--> 207         tendencies['explicit'] = self._compute_type('explicit')
    208         #  Tendencies due to implicit and adjustment processes need to be
    209         #  calculated from a state that is already adjusted after explicit stuff

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/process/time_dependent_process.py in _compute_type(self, proctype)
    253             if self.time['steps'] % step_ratio == 0:
    254                 proc.time['active_now'] = True
--> 255                 tenddict = proc.compute()
    256             else:
    257                 # proc.tendencies is unchanged from last subprocess timestep if we didn't recompute it above

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/process/time_dependent_process.py in compute(self)
    227                 self.tendencies[varname] += tend
    228         # Finally compute my own tendencies, if any
--> 229         self_tend = self._compute()
    230         #  Adjustment processes _compute method returns absolute adjustment
    231         #  Needs to be converted to rate of change

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/process/energy_budget.py in _compute(self)
     70 
     71     def _compute(self):
---> 72         tendencies = self._temperature_tendencies()
     73         return tendencies
     74 

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/process/energy_budget.py in _temperature_tendencies(self)
     58 
     59     def _temperature_tendencies(self):
---> 60         self._compute_heating_rates()
     61         tendencies = {}
     62         for varname, value in self.state.items():

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/radiation/rrtm/rrtmg_sw.py in _compute_heating_rates(self)
    195          inflgsw, iceflgsw, liqflgsw,
    196          cldfrac, ciwp, clwp, reic, relq, tauc, ssac, asmc, fsfc,
--> 197          tauaer, ssaaer, asmaer, ecaer,) = self._prepare_sw_arguments()
    198         if icld == 0:  # clear-sky only
    199             cldfmcl = np.zeros((ngptsw,ncol,nlay))

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/radiation/rrtm/rrtmg_sw.py in _prepare_sw_arguments(self)
    169         # Aerosol optical depth at 0.55 micron (iaer=6 only), Dimensions,  (ncol,nlay,naerec)] #  (non-delta scaled)
    170         # wenshanw
--> 171         ecaer = np.transpose(_climlab_to_rrtm(self.ecaer * np.ones_like(self.Tatm)) * np.ones([naerec,ncol,nlay]), (1,2,0))
    172         #ecaer = np.transpose(self.ecaer[..., ::-1] * np.ones([naerec,ncol,nlay]), (1,2,0))
    173         print("rrtmg_sw.py")

~/anaconda2/envs/rigb/lib/python3.6/site-packages/climlab/radiation/rrtm/utils.py in _climlab_to_rrtm(field)
     83         return field
     84     elif len(shape) > 2:
---> 85         raise ValueError('lat-lon grids not yet supported here.')
     86     #elif len(shape)==3:  # (num_lat, num_lon, num_lev)
     87         #  Need to reshape this array

ValueError: lat-lon grids not yet supported here.
```
I figured the `_climlab_to_rrtm` function probably takes this 3-D array as (lat, lon, layer) rather than (types_of_aerosols, column, layer).

Temporary fix:
In rrtmg_sw.py, change
`ecaer = np.transpose(_climlab_to_rrtm(self.ecaer * np.ones_like(self.Tatm)) * np.ones([naerec,ncol,nlay]), (1,2,0))`
to 
`ecaer = np.transpose(self.ecaer[..., ::-1] * np.ones([naerec,ncol,nlay]), (1,2,0))`

Thank you!

**Comment by wenshanw** [comment]
I don't have strong opinions over the order. Maybe the easiest way for coding?

**Comment by brian-rose** [comment]
This is finally resolved as of #198
